go/dism/dism.go

```diff
@@ -147,17 +153,31 @@ func (s Session) RemovePackage(
 	cancelEvent *windows.Handle,
 	progressCallback unsafe.Pointer,
 ) error {
-	return DismRemovePackage(s.Handle, helpers.StringToPtrOrNil(identifier), packageIdentifier, cancelEvent, progressCallback, nil)
+	return s.checkError(DismRemovePackage(*s.Handle, helpers.StringToPtrOrNil(identifier), packageIdentifier, cancelEvent, progressCallback, nil))
 }
 
 // Close closes the session and shuts down dism. This must be called prior to exiting.
 func (s Session) Close() error {
-	if err := DismCloseSession(s.Handle); err != nil {
+	if err := DismCloseSession(*s.Handle); err != nil {
 		return err
 	}
 	return DismShutdown()
 }
 
+// checkError validates the error returned by DISM API and reloads the session if needed
+func (s Session) checkError(err error) error {
+	if err == DISMAPI_S_RELOAD_IMAGE_SESSION_REQUIRED {
+		_ = DismCloseSession(*s.Handle)
```

Comment:
The return value here should probably be handled and possibly logged for debugging purposes, even if we don't return it from the function.

Reply:
handled the error and logging it